packages/h5p-server/src/ContentUserDataManager.ts

```diff
@@ -0,0 +1,154 @@
+     * @param contentId The id of the content to load user data from
+     * @param dataType Used by the h5p.js client
+     * @param subContentId The id provided by the h5p.js client call
+     * @param user The user who owns this object
+     * @returns the saved state as string
+     */
+    public async loadContentUserData(
+        contentId: ContentId,
+        dataType: string,
+        subContentId: string,
+        user: IUser
+    ): Promise<string> {
+        if (!this.contentUserDataStorage) {
+            return undefined;
+        }
+
+        if (!this.contentUserDataStorage.loadContentUserData) {
+            log.warn(
+                `the loadContentUserData method is not implemented in the provided contentUserDataStorage.`
+            );
+            return undefined;
+        }
+
+        log.info(
+            `loading contentUserData for user with id ${user.id} and contentId ${contentId}`
+        );
+
+        return this.contentUserDataStorage.loadContentUserData(
+            contentId,
+            dataType,
+            subContentId,
+            user
+        );
+    }
+
+    /**
+     * Loads the contentUserData for given contentId and user. The returned data is an array of IContentUserData where the position in the array corresponds with the subContentId or undefined if there is no contentUserData.
+     *
+     * @param contentId The id of the content to load user data from
+     * @param user The user who owns this object
+     * @returns an array of IContentUserData or undefined if no contentUserData is found.
+     */
+    public async generateContentUserDataIntegration(
+        contentId: ContentId,
+        user: IUser
+    ): Promise<IContentUserData[]> {
+        log.info(
+            `generating contentUserDataIntegration for user with id ${user.id} and contentId ${contentId}`
+        );
+
+        if (!this.contentUserDataStorage) {
+            return undefined;
+        }
+
+        const states = await this.contentUserDataStorage.listByContent(
+            contentId,
+            user.id
+        );
+
+        const sortedStates = states.sort(
+            (a, b) => Number(a.subContentId) - Number(b.subContentId)
+        );
+
+        const mappedStates = sortedStates.map((state) => ({
+            [state.dataType]: state.userState
+        }));
+
+        return mappedStates;
+    }
+
+    /**
+     * Saves the contentUserData for given contentId, dataType and subContentId
+     * @param contentId The id of the content to load user data from
+     * @param dataType Used by the h5p.js client
+     * @param subContentId The id provided by the h5p.js client call
+     * @param userState The userState as string
+     * @param user The user who owns this object
+     * @returns the saved state as string
+     */
+    public async saveContentUserData(
+        contentId: string,
+        dataType: string,
+        subcontentId: string,
+        userState: string,
+        invalidate: 0 | 1,
```

Comment:
boolean would be better here; the library doesn't use 0 or 1 for booleans anywhere else, except if it passes values to the client

Reply:
I know, but it is what the core sends. But I guess changing it to boolean would not be a problem.

Reply:
The paramters of the core are already changed in other places. We simply have to map the values to the internal types and back. The core is very inconsistent in how booleans are used (0, "0" or false are used) and it's better to have consistence within the library.